src/server/system_services/master_key_manager.js

```diff
@@ -170,14 +213,16 @@ class MasterKeysManager {
     /**
      * 
      * @param {String} m_key_id 
-     * @param {SensitiveString} new_root_key 
      * @returns {Object}  
      */
-    _reencrypt_master_key_by_root(m_key_id, new_root_key) {
+    _reencrypt_master_key_by_current_root(m_key_id, new_root_key_id) {
         const decrypted_mkey = this.get_master_key_by_id(m_key_id); // returns the decrypted m_key
         if (!decrypted_mkey) throw new Error('NO_SUCH_KEY');
-        this._update_root_key(new_root_key.unwrap()); // update ROOT_KEY with new secret provided
-        return this.encrypt_buffer_with_master_key_id(decrypted_mkey.cipher_key, ROOT_KEY);
+        const encrypted_key = this.encrypt_buffer_with_master_key_id(decrypted_mkey.cipher_key,
+            new_root_key_id);
+        decrypted_mkey.root_key_id = new_root_key_id;
+        decrypted_mkey.master_key_id = undefined;
```

Comment:
again why unsetting this property? if this key's master is root key why don't you want to set it to the new one? maybe root_key_id == master_key_id ?